lib/pbench/server/api/resources/query_apis/index_mappings.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+from flask import jsonify
+from logging import Logger
+from typing import Any, AnyStr, Dict
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources.query_apis import ElasticBase, Schema
+
+
+class IndexMappings(ElasticBase):
+    """
+    Get properties of the run-data document type on which we can search.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(config, logger, Schema())
+
+    def assemble(self, json_data: Dict[AnyStr, Any]) -> Dict[AnyStr, Any]:
+        """
+        Get properties of the run-data document for the index containing the current month
+        """
+        current_month = datetime.strptime(datetime.today().strftime("%Y-%m"), "%Y-%m")
+        uri_fragment = self._gen_month_range("run", current_month, current_month)
+        self.logger.info("index mapping key: {}", uri_fragment)
+        return {"path": f"/{uri_fragment}/_mapping", "kwargs": {}}
+
+    def postprocess(self, es_json: Dict[AnyStr, Any]) -> Dict[AnyStr, Any]:
+        """
+        Returns a dictionary of run documents properties.
+        """
+        index = list(es_json.keys())[0]
+        mappings = es_json[index]["mappings"]
+        self.logger.info(mappings)
+        result = {}
+        for property in mappings["properties"]:
+            self.logger.info(property)
+            if "properties" in mappings["properties"][property]:
+                result[property] = list(
+                    mappings["properties"][property]["properties"].keys()
+                )
+
+        result.pop("authorization", None)
```

Comment:
Huh; that's an interesting point. The actual search will control access to datasets, though, so I'm not sure it's necessary. This introduces the idea of "Pbench metadata" fields we're more or less hiding from a client. Is that a direction we want to go? The dashboard may want to focus the search on features of the run, but possibly it'd be more appropriate for the dashboard effect to remove this if it doesn't want that to appear in the search dashboard setup pane?

Reply:
yeah I am not sure either, I thought this field is not useful to the dashboard anyway so why to return, but I also probably agree that it's a dashboard's decision whether to add this on search pane. However, even if it adds this on the pane how would that be useful because the search is anyway gonna be user access controlled.